transaction/context.go

```diff
@@ -0,0 +1,180 @@
+package transaction
+
+import (
+	"context"
+	"database/sql"
+	"runtime/debug"
+
+	"gitlab.cee.redhat.com/service/uhc-clusters-service/pkg/global"
+
+	errors "github.com/zgalor/weberr"
+)
+
+type contextKey int
+
+const (
+	transactionKey contextKey = iota
+)
+
+// NewContext returns a new context with transaction stored in it
+func NewContext(ctx context.Context) (context.Context, error) {
+	tx, err := new()
+	if err != nil {
+		return nil, err
+	}
+
+	return context.WithValue(ctx, transactionKey, tx), nil
+}
+
+// TxContext creates a new transaction context from context.Background()
+func TxContext() (ctx context.Context, err error) {
```

Comment:
I think we should have a public `Transaction` type. This function should not create that transaction, it should only put it in the context. It should then be named `IntoContext` and shouldn't return an error. For example:

```go
// Create the unit of work:
var transaction.Unit unit = manager.Begin(ctx)

// Put it into the context:
ctx = transaction.IntoContext(ctx, unit)
```